storm-core/src/jvm/backtype/storm/messaging/netty/Client.java

```diff
@@ -317,7 +317,7 @@ private int iteratorSize(Iterator<TaskMessage> msgs) {
      * If the write operation fails, then we will close the channel and trigger a reconnect.
      */
     private void flushMessages(Channel channel, final MessageBatch batch) {
-        if(batch.isEmpty()){
+        if(null == batch || batch.isEmpty()){
```

Comment:
style nits:
* `if(` -> `if (`
* `){` -> `) {`

Those are more consistent with the rest of this file.